x-pack/plugins/security/server/routes/api/public/roles/put.js

```diff
@@ -84,30 +79,30 @@ export function initPutRolesApi(
     }, identity);
   };
 
-  const getGlobalSchema = () => {
+  const getKibanaSchema = () => {
     const privileges = authorization.privileges.get();
-    const featureObject = Object.entries(privileges.features).reduce((acc, [feature, featurePrivileges]) => ({
-      ...acc,
-      [feature]: Joi.array().items(Joi.string().valid(Object.keys(featurePrivileges)))
-    }), {});
-    const featureSchema = Joi.object(featureObject);
-    return Joi.object({
-      minimum: Joi.array().items(Joi.string().valid(Object.keys(privileges.global))),
-      feature: featureSchema,
+    const allSpacesSchema = Joi.array().length(1).items(Joi.string().valid([GLOBAL_RESOURCE]));
+    return Joi.array().items(
+      Joi.object({
+        base: Joi.alternatives().when('spaces', {
+          is: allSpacesSchema,
+          then: Joi.array().items(Joi.string().valid(Object.keys(privileges.global))),
+          otherwise: Joi.array().items(Joi.string().valid(Object.keys(privileges.space))),
+        }),
+        feature: Object.entries(privileges.features).reduce((acc, [feature, featurePrivileges]) => ({
+          ...acc,
+          [feature]: Joi.array().items(Joi.string().valid(Object.keys(featurePrivileges)))
```

Comment:
I'm not sure we want to fail for unrecognized features/feature privileges. Consider the following:
1) User creates role, specifying privileges for the "my_nifty" feature
2) User disables "my_nifty" feature, restarts Kibana
3) User attempts to update role

Unless we also filter out unknown features when retrieving (which I don't think we want to do), then updating this role won't be possible through the UI

Reply:
I can remove this while I'm in here, it was there previously and I was just moving it forward.